# 1.- PRIMEROS PASOS

# Importamos las Librerías

In [993]:
import numpy as np
import pandas as pd

* Recordatorio: De implementarse necesariamente cambios en nuestro "Dataframe", lo mejor es trabajar con una copia 

# Carga de Archivo CSV

In [994]:
#Leemos el csv
df_ = pd.read_csv("actividad.csv")
#df_ es nuestro dataframe original!!

In [995]:
#Vizualizamos una pequeña muestra
df_.sample()

Profile Name           Start Time  Duration Attributes  \
5970       Yakomo  2022-02-06 00:34:19  00:19:16        NaN   

                                                            Title  \
5970  BAKI: La saga del gran torneo de Raitai: Kaioh (Capítulo 7)   

     Supplemental Video Type                               Device Type  \
5970                     NaN  Sony Sony Android TV 2018 M2 4K Smart TV   

      Bookmark Latest Bookmark    Country  
5970  00:22:25        00:22:25  PE (Peru)

# Repaso de la Definición de Cada Variable en el "DATAFRAME"

* Profile Name: Se basa en el nombre del perfil, conociendo que una cuenta puede ostentar de varios. Posiblemente debamos estar al tanto de la actividad de cada uno por separado, ya que el dueño de cada perfil es un usuario completamente diferente. -lo podemos cambiar por "profile"-

* Star Time: Guarda el Tiempo exacto de inicio al reproducirse un determinado título de la plataforma -lo podemos renombrar como "time_start"-

* Duration: Nos muestra el tiempo de visualización, a partir del tiempo que iniciamos un título hasta que lo damos por finalizado. -lo podemos renombrar "duration"-

* Atributes: Alguna peculiaridad añadida a la hora de visualizar el título, en este parece referenciar a la accción de "Autoreproducción" -lo podemos renombrar como "atributes"-

* Title: Se refiere al título del respectivo material audiovisual que se ha reproducido. -lo podemos renombrar como "title"-

* Suplemental Video Type: Visualizando los datos por encima, vemos que identifica si algún material reproducido funciona como sumplemeto a un material mucho más robusto. Por ejemplo puede ser un Trailer o un Teaser, en términos generales avances o contenidos que complementan la experiencia. - lo podemos llamar como "suple_type"

* Device Type: Guarda exactamente desde que tipo de dispositivo se tuvo lugar la respectiva reproducción. -lo podemos llamar como "device"-

* Para propósitos de un Análisis por el momento básico descartaremos las 3 últimas columnas, las de bookmark porque no consideremos que aporten gran valor y la  del país donde se provocó la reproducción, dado a que el 100% de los datos se obtuvieron en Perú. Aunque igualmente es un dato que no deberíamos olvidar. 

# Procedemos a renombrar las columnas como lo estipulamos
# También eliminaremos las 3 últimas columnas ["Bookmark","Latest Bookmark","Country"]

In [996]:
#Creamos la copia de nuestro DataFrame
df = df_.copy()

In [997]:
#Procedemos a elminar las 3 últimas columnas
df.drop(["Bookmark","Latest Bookmark","Country"], axis=1,inplace=True)

* Nota: Normalmente se renombran las columnas para facilitar nuestro trabajo al digitar los nombres de las variables
* Para esto no debemos olvidar las buenas prácticas a la hora de nombrar una variable (en este caso columna)

In [998]:
#De esta forma renombramos nuestras columnas
df.columns = ["profile","time_start","duration","atributes","title","suple_type","device"]

In [999]:
df

profile           time_start  duration  \
0        Kids  2019-12-28 14:14:17  00:13:33   
1        Kids  2018-12-10 18:48:27  00:16:49   
2        Kids  2018-12-06 02:12:19  00:08:11   
3        Kids  2018-12-06 01:45:37  00:26:20   
4        Kids  2018-12-06 01:35:35  00:01:13   
5        Kids  2018-12-06 01:12:40  00:22:49   
6        Kids  2018-12-06 00:45:57  00:22:50   
7        Kids  2018-12-06 00:22:53  00:22:50   
...       ...                  ...       ...   
14854  Yakomo  2018-05-26 22:23:29  00:27:26   
14855  Yakomo  2018-05-26 22:18:49  00:04:33   
14856  Yakomo  2018-05-23 21:31:51  00:03:02   
14857  Yakomo  2018-05-23 21:21:51  00:02:49   
14858  Yakomo  2018-05-23 21:04:49  00:14:54   
14859  Yakomo  2018-05-23 21:03:12  00:01:32   
14860  Yakomo  2018-05-23 20:42:08  00:20:54   
14861  Yakomo  2018-05-23 20:41:19  00:00:02   

                                    atributes  \
0      Autoplayed: user action: Unspecified;    
1                                         NaN   
2             Autoplayed: user action: None;    
3      Autoplayed: user action: Unspecified;    
4             Autoplayed: user action: None;    
5             Autoplayed: user action: None;    
6             Autoplayed: user action: None;    
7             Autoplayed: user action: None;    
...                                       ...   
14854  Autoplayed: user action: Unspecified;    
14855  Autoplayed: user action: Unspecified;    
14856  Autoplayed: user action: Unspecified;    
14857                                     NaN   
14858  Autoplayed: user action: Unspecified;    
14859                                     NaN   
14860                                     NaN   
14861         Autoplayed: user action: None;    

                                                                                                     title  \
0                                                                                            Kung Fu Panda   
1                                                                        Toy Story: olvidados en el tiempo   
2      Pocoyó: Temporada 1: ¡Busca, Loula, busca! / La pequeña nube / Un regalo para Elly / El baile de...   
3      Pocoyó: Temporada 1: Un día de lluvia / ¡Música, maestro! / Juego limpio / Riing riing (Capítulo 1)   
4                                   La casa de Mickey Mouse: Temporada 2: ¡Pluto al rescate! (Capítulo 21)   
5                              La casa de Mickey Mouse: Temporada 2: Daisy, la espía secreta (Capítulo 20)   
6                La casa de Mickey Mouse: Temporada 2: El día de Acción de Gracias de Mickey (Capítulo 19)   
7                           La casa de Mickey Mouse: Temporada 2: El bobo problema de Mickey (Capítulo 18)   
...                                                                                                    ...   
14854                                                                       Jurassic World: Mundo Jurásico   
14855                                                                       Jurassic World: Mundo Jurásico   
14856                                          Altered Carbon: Temporada 1: Retorno al pasado (Capítulo 1)   
14857                                 Sword Art Online: Sword Art Online II: Un pequeño paso (Capítulo 14)   
14858                                  Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14859                                  Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14860                                   Sword Art Online: Sword Art Online II: Bala fantasma (Capítulo 12)   
14861                            Sword Art Online: Sword Art Online II: El mundo de las armas (Capítulo 1)   

      suple_type                                        device  
0            NaN      Sony Sony Android TV 2018 M2 4K Smart TV  
1            NaN  Android DefaultWidevineL3Phone Android Phone  
2            NaN                        Samsung 2014 MStar DTV  
3            NaN 

# Realizando una pequeña revisión "profile"

* Dado que en la variable profile (perfiles) se encuentran datos de otras personas, no deseo vulnerar la privacidad de terceros por lo cual solo enfocaré este análisis a mi persona. 
* Para llevar a cabo ello, lo mejor sería visualizar qué perfiles están involucrados en la tabla..

In [1000]:
#Cuales son esos perfiles?
df["profile"].value_counts()

Yakomo    9140
Y         5684
Kids        38
Name: profile, dtype: int64

# Conclusión de la Revisión

* Los perfiles involucrados son 3:
    * El que le pertenece a mi persona "Yakomo"
    * Un segundo denominado "Y" para proteger su privacidad
    * Un Tercero llamado Kids, perfil que contiene protección parental.

* Como este análisis esta dirigido a analizar el comportamiento de mi perfil personal procederemos a descartar aquellos perfiles que no tengan la denominación "Yakomo" 
* O mejor aún, solo trabajemos con los datos que aludan al nombre de mi propio perfil ¿Cómo lo Haremos?

In [1001]:
#Creamos una condición que signifique [Pefil es Yakomo]
condicion = df["profile"] == "Yakomo"
#Teniendo la condición lista solo la debemos invocar de la siguiente manera. 
df[condicion]


profile           time_start  duration  \
5722   Yakomo  2022-04-11 18:38:28  00:13:25   
5723   Yakomo  2022-04-11 18:13:46  00:20:58   
5724   Yakomo  2022-04-11 13:41:14  00:19:31   
5725   Yakomo  2022-04-11 13:32:44  00:08:26   
5726   Yakomo  2022-04-11 13:32:15  00:00:18   
5727   Yakomo  2022-04-11 13:32:07  00:00:05   
5728   Yakomo  2022-04-10 23:51:48  00:14:37   
5729   Yakomo  2022-04-10 23:51:36  00:00:09   
...       ...                  ...       ...   
14854  Yakomo  2018-05-26 22:23:29  00:27:26   
14855  Yakomo  2018-05-26 22:18:49  00:04:33   
14856  Yakomo  2018-05-23 21:31:51  00:03:02   
14857  Yakomo  2018-05-23 21:21:51  00:02:49   
14858  Yakomo  2018-05-23 21:04:49  00:14:54   
14859  Yakomo  2018-05-23 21:03:12  00:01:32   
14860  Yakomo  2018-05-23 20:42:08  00:20:54   
14861  Yakomo  2018-05-23 20:41:19  00:00:02   

                                         atributes  \
5722                                           NaN   
5723   Autoplayed: user action: User_Interaction;    
5724                                           NaN   
5725                                           NaN   
5726               Autoplayed: user action: None;    
5727               Autoplayed: user action: None;    
5728                                           NaN   
5729               Autoplayed: user action: None;    
...                                            ...   
14854       Autoplayed: user action: Unspecified;    
14855       Autoplayed: user action: Unspecified;    
14856       Autoplayed: user action: Unspecified;    
14857                                          NaN   
14858       Autoplayed: user action: Unspecified;    
14859                                          NaN   
14860                                          NaN   
14861              Autoplayed: user action: None;    

                                                                           title  \
5722      La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
5723      La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
5724      La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
5725      La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
5726                         Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   
5727                                Better Call Saul: Season 3_hook_primary_16x9   
5728                                                          Regreso al espacio   
5729                                                 Tráiler: Regreso al espacio   
...                                                                          ...   
14854                                             Jurassic World: Mundo Jurásico   
14855                                             Jurassic World: Mundo Jurásico   
14856                Altered Carbon: Temporada 1: Retorno al pasado (Capítulo 1)   
14857       Sword Art Online: Sword Art Online II: Un pequeño paso (Capítulo 14)   
14858        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14859        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14860         Sword Art Online: Sword Art Online II: Bala fantasma (Capítulo 12)   
14861  Sword Art Online: Sword Art Online II: El mundo de las armas (Capítulo 1)   

      suple_type                                    device  
5722         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5723         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5724         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5725         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5726        HOOK  Sony Sony Android TV 2018 M2 4K Smart TV  
5727        HOOK  Sony Sony Android TV 2018 M2 4K Smart TV  
5728         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5729     TRAILER  Sony Sony Android TV 2018 M2 4K Smart TV  
...          ...                                       ...  
14854        NaN                             

* Listo ahora solo nos queda reemplazar nuestro dataframe de trabajo para que solo trabajemos con lo obtenido

In [1002]:
#Estamos casi listos para trabajar
df = df[condicion]

In [1003]:
#Como vemos el dataframe ahora solo contiene un perfil
df["profile"].value_counts()

Yakomo    9140
Name: profile, dtype: int64

In [1004]:
df

profile           time_start  duration  \
5722   Yakomo  2022-04-11 18:38:28  00:13:25   
5723   Yakomo  2022-04-11 18:13:46  00:20:58   
5724   Yakomo  2022-04-11 13:41:14  00:19:31   
5725   Yakomo  2022-04-11 13:32:44  00:08:26   
5726   Yakomo  2022-04-11 13:32:15  00:00:18   
5727   Yakomo  2022-04-11 13:32:07  00:00:05   
5728   Yakomo  2022-04-10 23:51:48  00:14:37   
5729   Yakomo  2022-04-10 23:51:36  00:00:09   
...       ...                  ...       ...   
14854  Yakomo  2018-05-26 22:23:29  00:27:26   
14855  Yakomo  2018-05-26 22:18:49  00:04:33   
14856  Yakomo  2018-05-23 21:31:51  00:03:02   
14857  Yakomo  2018-05-23 21:21:51  00:02:49   
14858  Yakomo  2018-05-23 21:04:49  00:14:54   
14859  Yakomo  2018-05-23 21:03:12  00:01:32   
14860  Yakomo  2018-05-23 20:42:08  00:20:54   
14861  Yakomo  2018-05-23 20:41:19  00:00:02   

                                         atributes  \
5722                                           NaN   
5723   Autoplayed: user action: User_Interaction;    
5724                                           NaN   
5725                                           NaN   
5726               Autoplayed: user action: None;    
5727               Autoplayed: user action: None;    
5728                                           NaN   
5729               Autoplayed: user action: None;    
...                                            ...   
14854       Autoplayed: user action: Unspecified;    
14855       Autoplayed: user action: Unspecified;    
14856       Autoplayed: user action: Unspecified;    
14857                                          NaN   
14858       Autoplayed: user action: Unspecified;    
14859                                          NaN   
14860                                          NaN   
14861              Autoplayed: user action: None;    

                                                                           title  \
5722      La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
5723      La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
5724      La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
5725      La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
5726                         Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   
5727                                Better Call Saul: Season 3_hook_primary_16x9   
5728                                                          Regreso al espacio   
5729                                                 Tráiler: Regreso al espacio   
...                                                                          ...   
14854                                             Jurassic World: Mundo Jurásico   
14855                                             Jurassic World: Mundo Jurásico   
14856                Altered Carbon: Temporada 1: Retorno al pasado (Capítulo 1)   
14857       Sword Art Online: Sword Art Online II: Un pequeño paso (Capítulo 14)   
14858        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14859        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14860         Sword Art Online: Sword Art Online II: Bala fantasma (Capítulo 12)   
14861  Sword Art Online: Sword Art Online II: El mundo de las armas (Capítulo 1)   

      suple_type                                    device  
5722         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5723         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5724         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5725         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5726        HOOK  Sony Sony Android TV 2018 M2 4K Smart TV  
5727        HOOK  Sony Sony Android TV 2018 M2 4K Smart TV  
5728         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5729     TRAILER  Sony Sony Android TV 2018 M2 4K Smart TV  
...          ...                                       ...  
14854        NaN                             

# 2.- FASE DE ANÁLISIS EXPLORATORIO


* Una vez a grandes rasgos hemos obtenido la tabla o "dataframe" con el cual trabajaremos, es momento de analizar detenidamente nuestros dataframe y con ello sus variables. 
* ¿Son todas necesarias? ¿Tienen un Formato Adecuado? ¿Son del tipo de Dato Adecuado?
* Esas son algunas preguntas de las cuales debemos responder

Empecemos por el Dataframe como tal...

* ¿Cuál es su dimensión?

In [1005]:
df.shape

(9140, 7)

Respuesta: Contamos con un Dataframe de 9140 filas o registros, enfrentadas con 7 variables o columnas

* ¿Cuál es la información más resaltante en la tabla?

In [1006]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9140 entries, 5722 to 14861
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   profile     9140 non-null   object
 1   time_start  9140 non-null   object
 2   duration    9140 non-null   object
 3   atributes   5573 non-null   object
 4   title       9140 non-null   object
 5   suple_type  765 non-null    object
 6   device      9140 non-null   object
dtypes: object(7)
memory usage: 571.2+ KB


Respuesta: 
* Nombres de las columnas: profile,time_start,duration,atributes,title,suple_type,device
* Tipos de cada variable: Todas son Object ¿Deberían ser todas así? [un tipo dato object no es específico, pero suele tomarse como string]
* Importante!! Encontramos valores faltantes en los siguientes apartados: atributes y en suple_type [Algo normal dado a que son opcionales]
* En la mayoría de casos sería admisible que nuestras variables fueran objeto, sin embargo time_start podría trabajarse de manera mucho más profunda ¿Cómo?

# Trabajando la variable "time_start"

In [1007]:
#Repasemos como suele verse esta variable
df["time_start"]

5722     2022-04-11 18:38:28
5723     2022-04-11 18:13:46
5724     2022-04-11 13:41:14
5725     2022-04-11 13:32:44
5726     2022-04-11 13:32:15
5727     2022-04-11 13:32:07
5728     2022-04-10 23:51:48
5729     2022-04-10 23:51:36
                ...         
14854    2018-05-26 22:23:29
14855    2018-05-26 22:18:49
14856    2018-05-23 21:31:51
14857    2018-05-23 21:21:51
14858    2018-05-23 21:04:49
14859    2018-05-23 21:03:12
14860    2018-05-23 20:42:08
14861    2018-05-23 20:41:19
Name: time_start, Length: 9140, dtype: object

* Vemos que por un lado se muestra la fecha en la que se produce la reproducción y "separado por un espacio", de inmediato nos muestra la hora, minuto y segundo en el que se realiza la misma reproducción.
* Lo que nos hace pensar... es esto "conveniente?"

Según expertos lo mejor es siempre tener un manejo por fechas separado de cualquier otro valor, esto nos otorgará habilidades de análisis muchísimo más poderosas. 

In [1008]:
#Perfecto veamos qué hacer
df

profile           time_start  duration  \
5722   Yakomo  2022-04-11 18:38:28  00:13:25   
5723   Yakomo  2022-04-11 18:13:46  00:20:58   
5724   Yakomo  2022-04-11 13:41:14  00:19:31   
5725   Yakomo  2022-04-11 13:32:44  00:08:26   
5726   Yakomo  2022-04-11 13:32:15  00:00:18   
5727   Yakomo  2022-04-11 13:32:07  00:00:05   
5728   Yakomo  2022-04-10 23:51:48  00:14:37   
5729   Yakomo  2022-04-10 23:51:36  00:00:09   
...       ...                  ...       ...   
14854  Yakomo  2018-05-26 22:23:29  00:27:26   
14855  Yakomo  2018-05-26 22:18:49  00:04:33   
14856  Yakomo  2018-05-23 21:31:51  00:03:02   
14857  Yakomo  2018-05-23 21:21:51  00:02:49   
14858  Yakomo  2018-05-23 21:04:49  00:14:54   
14859  Yakomo  2018-05-23 21:03:12  00:01:32   
14860  Yakomo  2018-05-23 20:42:08  00:20:54   
14861  Yakomo  2018-05-23 20:41:19  00:00:02   

                                         atributes  \
5722                                           NaN   
5723   Autoplayed: user action: User_Interaction;    
5724                                           NaN   
5725                                           NaN   
5726               Autoplayed: user action: None;    
5727               Autoplayed: user action: None;    
5728                                           NaN   
5729               Autoplayed: user action: None;    
...                                            ...   
14854       Autoplayed: user action: Unspecified;    
14855       Autoplayed: user action: Unspecified;    
14856       Autoplayed: user action: Unspecified;    
14857                                          NaN   
14858       Autoplayed: user action: Unspecified;    
14859                                          NaN   
14860                                          NaN   
14861              Autoplayed: user action: None;    

                                                                           title  \
5722      La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
5723      La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
5724      La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
5725      La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
5726                         Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   
5727                                Better Call Saul: Season 3_hook_primary_16x9   
5728                                                          Regreso al espacio   
5729                                                 Tráiler: Regreso al espacio   
...                                                                          ...   
14854                                             Jurassic World: Mundo Jurásico   
14855                                             Jurassic World: Mundo Jurásico   
14856                Altered Carbon: Temporada 1: Retorno al pasado (Capítulo 1)   
14857       Sword Art Online: Sword Art Online II: Un pequeño paso (Capítulo 14)   
14858        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14859        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14860         Sword Art Online: Sword Art Online II: Bala fantasma (Capítulo 12)   
14861  Sword Art Online: Sword Art Online II: El mundo de las armas (Capítulo 1)   

      suple_type                                    device  
5722         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5723         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5724         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5725         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5726        HOOK  Sony Sony Android TV 2018 M2 4K Smart TV  
5727        HOOK  Sony Sony Android TV 2018 M2 4K Smart TV  
5728         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  
5729     TRAILER  Sony Sony Android TV 2018 M2 4K Smart TV  
...          ...                                       ...  
14854        NaN                             

Se nos puede ocurrir separar los datos en como tal "date" y el propio "tiempo de inicio"

In [1009]:
#Creamos una lista "date" que se generara a partir de la serie df["time_start"]
date = [elemento.split(" ")[0] for elemento in df["time_start"]] #Con split creamos una lista con los elementos separados por (" ") usamos [0] ya que la fecha es el primer elemento
time_start = [elemento.split(" ")[1] for elemento in df["time_start"]]

Poseyendo las listas con los datos necesarios, procedemos a modificar y/o crear columnas según convenga.

In [1010]:
#Para esto usamos la función assign, donde recibe como parámetro ("nombre de la columna" = "objeto_iterable", en nuestro caso una lista)
df = df.assign(date = date,time_start=time_start)
#Si la columna existe, la reemplazará, de no existir se creará. 

In [1011]:
df

profile time_start  duration  \
5722   Yakomo   18:38:28  00:13:25   
5723   Yakomo   18:13:46  00:20:58   
5724   Yakomo   13:41:14  00:19:31   
5725   Yakomo   13:32:44  00:08:26   
5726   Yakomo   13:32:15  00:00:18   
5727   Yakomo   13:32:07  00:00:05   
5728   Yakomo   23:51:48  00:14:37   
5729   Yakomo   23:51:36  00:00:09   
...       ...        ...       ...   
14854  Yakomo   22:23:29  00:27:26   
14855  Yakomo   22:18:49  00:04:33   
14856  Yakomo   21:31:51  00:03:02   
14857  Yakomo   21:21:51  00:02:49   
14858  Yakomo   21:04:49  00:14:54   
14859  Yakomo   21:03:12  00:01:32   
14860  Yakomo   20:42:08  00:20:54   
14861  Yakomo   20:41:19  00:00:02   

                                         atributes  \
5722                                           NaN   
5723   Autoplayed: user action: User_Interaction;    
5724                                           NaN   
5725                                           NaN   
5726               Autoplayed: user action: None;    
5727               Autoplayed: user action: None;    
5728                                           NaN   
5729               Autoplayed: user action: None;    
...                                            ...   
14854       Autoplayed: user action: Unspecified;    
14855       Autoplayed: user action: Unspecified;    
14856       Autoplayed: user action: Unspecified;    
14857                                          NaN   
14858       Autoplayed: user action: Unspecified;    
14859                                          NaN   
14860                                          NaN   
14861              Autoplayed: user action: None;    

                                                                           title  \
5722      La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
5723      La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
5724      La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
5725      La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
5726                         Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   
5727                                Better Call Saul: Season 3_hook_primary_16x9   
5728                                                          Regreso al espacio   
5729                                                 Tráiler: Regreso al espacio   
...                                                                          ...   
14854                                             Jurassic World: Mundo Jurásico   
14855                                             Jurassic World: Mundo Jurásico   
14856                Altered Carbon: Temporada 1: Retorno al pasado (Capítulo 1)   
14857       Sword Art Online: Sword Art Online II: Un pequeño paso (Capítulo 14)   
14858        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14859        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
14860         Sword Art Online: Sword Art Online II: Bala fantasma (Capítulo 12)   
14861  Sword Art Online: Sword Art Online II: El mundo de las armas (Capítulo 1)   

      suple_type                                    device        date  
5722         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  2022-04-11  
5723         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  2022-04-11  
5724         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  2022-04-11  
5725         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  2022-04-11  
5726        HOOK  Sony Sony Android TV 2018 M2 4K Smart TV  2022-04-11  
5727        HOOK  Sony Sony Android TV 2018 M2 4K Smart TV  2022-04-11  
5728         NaN  Sony Sony Android TV 2018 M2 4K Smart TV  2022-04-10  
5729     TRAILER  Sony Sony Android TV 2018 M2 4K Smart TV  2022-04-10  
...          ...                                       ...         ...  
14854        NaN                                  Sony PS4  2018-05-26  
14855        NaN                

Como apreciación, vemos que en verdad tener la variable profile con un único valor no es que nos aporte mucha información. Solo debemos de tener en cuenta siempre a quien pertenece.

In [1012]:
#Procedemos a Despojarnos de  la columna perfil
df.drop(["profile"], axis=1,inplace=True) #Inplace = True, modifica nuestro dataframe sometido.
df.reset_index(drop=True,inplace=True) #Drop True es casi 100% necesario, nuevamente inplace = True

# Es momento de "Castear" el Dato Fecha

* Si bien tenemos el apartado fecha, recordemos que su tipo de dato posiblemente no corresponda al adecuado. 

In [1013]:
df.dtypes

time_start    object
duration      object
atributes     object
title         object
suple_type    object
device        object
date          object
dtype: object

In [1014]:
#Entonces transformaremos la columna date en una de formato fecha
df["date"] = pd.to_datetime(df["date"])

In [1015]:
#Corroboramos dicho Cambio
df.dtypes

time_start            object
duration              object
atributes             object
title                 object
suple_type            object
device                object
date          datetime64[ns]
dtype: object

In [1016]:
#Comprobamos nuestro data_frame
df.head(5)

time_start  duration                                    atributes  \
0   18:38:28  00:13:25                                          NaN   
1   18:13:46  00:20:58  Autoplayed: user action: User_Interaction;    
2   13:41:14  00:19:31                                          NaN   
3   13:32:44  00:08:26                                          NaN   
4   13:32:15  00:00:18              Autoplayed: user action: None;    

                                                                    title  \
0  La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
1  La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
2  La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
3  La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
4                     Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   

  suple_type                                    device       date  
0        NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
1        NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
2        NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
3        NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
4       HOOK  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11

# Visualizamos el contenido con Mucho más Detenemimiento, debemos encontrar PATRONES

* Por ahora dejaremos pasar las variables "time_start" y "duration" de seguro hay un método de como tratarlas, (averiguar)
* Pero que nos puede decir la variable atributes?

In [1017]:
df["atributes"].value_counts()

Autoplayed: user action: None;                                   2687
Autoplayed: user action: Unspecified;                            2505
Autoplayed: user action: User_Interaction;                        310
Has branched playback;                                             39
Has branched playback; Autoplayed: user action: Unspecified;       18
Has branched playback; Autoplayed: user action: None;              12
Autoplayed: user action: Visibility_Timeout;                        2
Name: atributes, dtype: int64

* Vemos que cada atributo contiene peculiarmente un ";" al final, lo que es inadecuado. Procedemos a repararlo:

In [1018]:
#Para esto creamos una función.
#Esto es recomendable ya que nuestra columna contiene valores faltantes. 
def mod_atributes(item): #Entonces recibimos un item
    if(type(item) == float): #Si este es de tipo float (el tipo de un dato None) 
        return item #No hacemos nigún cambio
    else: #de lo contrario
        item = item.strip("; ") #Eliminamos los caracteres que empañan nuestro dato
        return item #Siempre deberemos retornarlo. 

df["atributes"] = df["atributes"].apply(mod_atributes) #Aplicamos la funcion a nuestra columna correspondiente. 


In [1019]:
df

time_start  duration                                  atributes  \
0      18:38:28  00:13:25                                        NaN   
1      18:13:46  00:20:58  Autoplayed: user action: User_Interaction   
2      13:41:14  00:19:31                                        NaN   
3      13:32:44  00:08:26                                        NaN   
4      13:32:15  00:00:18              Autoplayed: user action: None   
5      13:32:07  00:00:05              Autoplayed: user action: None   
6      23:51:48  00:14:37                                        NaN   
7      23:51:36  00:00:09              Autoplayed: user action: None   
...         ...       ...                                        ...   
9132   22:23:29  00:27:26       Autoplayed: user action: Unspecified   
9133   22:18:49  00:04:33       Autoplayed: user action: Unspecified   
9134   21:31:51  00:03:02       Autoplayed: user action: Unspecified   
9135   21:21:51  00:02:49                                        NaN   
9136   21:04:49  00:14:54       Autoplayed: user action: Unspecified   
9137   21:03:12  00:01:32                                        NaN   
9138   20:42:08  00:20:54                                        NaN   
9139   20:41:19  00:00:02              Autoplayed: user action: None   

                                                                          title  \
0        La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
1        La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
2        La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
3        La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
4                           Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   
5                                  Better Call Saul: Season 3_hook_primary_16x9   
6                                                            Regreso al espacio   
7                                                   Tráiler: Regreso al espacio   
...                                                                         ...   
9132                                             Jurassic World: Mundo Jurásico   
9133                                             Jurassic World: Mundo Jurásico   
9134                Altered Carbon: Temporada 1: Retorno al pasado (Capítulo 1)   
9135       Sword Art Online: Sword Art Online II: Un pequeño paso (Capítulo 14)   
9136        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
9137        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
9138         Sword Art Online: Sword Art Online II: Bala fantasma (Capítulo 12)   
9139  Sword Art Online: Sword Art Online II: El mundo de las armas (Capítulo 1)   

     suple_type                                    device       date  
0           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
1           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
2           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
3           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
4          HOOK  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
5          HOOK  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
6           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-10  
7       TRAILER  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-10  
...         ...                                       ...        ...  
9132        NaN                                  Sony PS4 2018-05-26  
9133        NaN                                  Sony PS4 2018-05-26  
9134        NaN                                  Sony PS4 2018-05-23  
9135        NaN                                  Sony PS4 2018-05-23  
9136        NaN                                  Sony PS4 2018-05-23  
9137        NaN                                  Sony PS4 2018-05-23  
9138        NaN                                  Sony PS4 2018-05-23  
9139  

* Que más observamos de la columna atributos?

In [1020]:
df["atributes"].value_counts()

Autoplayed: user action: None                                  2687
Autoplayed: user action: Unspecified                           2505
Autoplayed: user action: User_Interaction                       310
Has branched playback                                            39
Has branched playback; Autoplayed: user action: Unspecified      18
Has branched playback; Autoplayed: user action: None             12
Autoplayed: user action: Visibility_Timeout                       2
Name: atributes, dtype: int64

* Observamos que existen dos tipos de atributos principales, la "Autoreproducción" que puede ser con acción, "nula","inespecífica","interación del usuario"
* Y el segundo atributo es la presencia de una Reproducción "Bifurcada" que como vemos suele contener al anterior atributo. 
* La "Autoreproducción" suele separarse con ":"
* Mientras que la Reproducción "Bifurcafa" se presenta con un ";" para denotarse y consecuentemente tiene el mismo formato que la "Autoreproducción"

* Qué observamos de la columna title?

In [1021]:
df["title"].sample(20)

8820                        Ultimate Spider-Man: Temporada 1: Por qué odio la educación física (Capítulo 6)
5506    Johnny Test: Temporada 6: Johnny contra el Dukenator / El zoológico infantil de Johnny (Capítulo...
4161                                      Vis a vis: Temporada 1: A las 5 en punto de la tarde (Capítulo 7)
1165                                                    Breaking Bad: Temporada 5: Ozymandias (Capítulo 14)
124                                         Rick y Morty: Temporada 4: El rickgreso del Jerri (Capítulo 10)
2303                                  That '70s Show: Temporada 7: Can't You Hear Me Knocking (Capítulo 13)
5323                                                    La peor bruja: Temporada 3: Magia mala (Capítulo 7)
3135                                        El final del paraíso: Temporada 1: La pelea final (Capítulo 87)
                                                       ...                                                 
4526                        

* Observamos una estructura carácterística, al menos en lo que respecta a series podemos decir que en un principio cada término esta separado por un ":" donde se empieza por el Título, le continua la temporada actual de visualización, para finalmente llegar con el Nombre del episodio. 
* Lamentablemenete el nombre de un episodio puede contener este caracter ":" lo cual podría entorpecer un futuro procecimiento. 
* Las películas por otra parte parecen no contener este formato, aunque seguro alguno de ellas contiene ":" en su nombre, no parecen contar con una estructura que podamos seguir. 
* Analizando mucho más también encontramos materiales que pueden pertenecer a otro grupo de materiales, ya sean los llamamos Hooks, trailers, teaser, etc.
* Pero para ello mejor visualizaremos a detenimiento la variable "suple_type" 

* Qué observamos de la columna suple_type?

In [1022]:
df["suple_type"].value_counts()

HOOK              444
TRAILER           274
TEASER_TRAILER     32
RECAP               6
PROMOTIONAL         5
BONUS_VIDEO         3
MUSIC_VIDEO         1
Name: suple_type, dtype: int64

* Interesante, la columna suple_type cuenta con datos peculiares.
* En primera este puede referir al material como HOOK, que vendría siendo el intento de netflix por engancharnos tras mostrarnos un pequeño fragmento. 
* Observamos la posiblidad que sea un Trailer, un Teaser_Trailer, una recapitulación, un promotional, un bonus_video, o también un musical. 
* Esto es genial, ahora sabemos que datos debemos catalogar únicamente como series o películas, EXACTO! aquellos con un atributo "NONE"
* ¿Cuántos serán?

Sabemos de antemano que suple_type contienen datos faltanes... pues es momento de contarlos

In [1023]:
df["suple_type"].isnull().sum()

8375

* Como resultado tenemos 8375 datos faltantes, aquellos que podremos catalogarlos como series o películas (Con esto consideraremos a materiales alternativos como documentales, meramente como películas)
* Ahora tenemos una mejor idea para ir catalogando nuestros títulos. 

* Qué observamos de la columna device?

In [1024]:
df["device"].value_counts()

Sony Sony Android TV 2018 M2 4K Smart TV          3803
Sony PS4                                          3617
Android DefaultWidevineL3Phone Android Phone       578
Chrome PC (Cadmium)                                493
Android DefaultWidevineL3Tablet Android Tablet     190
LG 2017 LG M16p Standard UHD TV Smart TV           179
Netflix Windows App - Cadmium Windows Mobile       159
Samsung 2018 Kant-M2 UHD TV (1.5G) Smart TV         59
DefaultWidevineAndroidPhone                         47
Samsung 2014 MStar DTV                              10
Samsung 2013 MStar X12 Smart TV                      4
LG 2018 Mstar LM18A STD UHD TV Smart TV              1
Name: device, dtype: int64

* En device reconoceremos los dispositivos empleados a la hora de visualizar un título en Netlix. 
* Posiblemente estos datos lo podríamos resumir según la marca del dispositivo o el tipo de dispositivo en sí, (ya sea de mesa o portatil) pero eso convendrá según el análisis. 

Finalmente la Fecha no resiste análisis por lo menos en este momento

In [1025]:
df.head(5)

time_start  duration                                  atributes  \
0   18:38:28  00:13:25                                        NaN   
1   18:13:46  00:20:58  Autoplayed: user action: User_Interaction   
2   13:41:14  00:19:31                                        NaN   
3   13:32:44  00:08:26                                        NaN   
4   13:32:15  00:00:18              Autoplayed: user action: None   

                                                                    title  \
0  La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
1  La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
2  La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
3  La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
4                     Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   

  suple_type                                    device       date  
0        NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
1        NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
2        NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
3        NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  
4       HOOK  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11

In [1026]:
#Separamos en una columna los titulos, formando una lista separada por nuestro caracter ":" el cual tiene incidencias para identificar una serie. 
df_sep_lista = df["title"].str.split(pat = ":", expand = False).to_frame() #colocamos el caracter, y NO lo expandimos, convirtiendo la serie asi en un dataframe

In [1027]:
df_sep_lista.sample(17)

title
7449                                          [MERLÍN,  Temporada 3,  El castillo de Fyrien (Capítulo 7)]
643                                                                [Oats Studios,  Volume 1_hook_02_16x9]
32                                         [The Disastrous Life of Saiki K.,  Season 1_hook_primary_16x9]
7972               [Pokémon,  Sol y Luna,  Sol y Luna,  ¡Sol, sustos y nueva base secreta! (Capítulo 12)]
5125                   [Lego,  Nexo Knights,  Temporada 1,  El libro de monstruos (Parte 2) (Capítulo 2)]
9138                               [Sword Art Online,  Sword Art Online II,  Bala fantasma (Capítulo 12)]
5447                   [Lego,  Nexo Knights,  Temporada 1,  El libro de monstruos (Parte 1) (Capítulo 1)]
4371                                        [Bob Esponja,  Temporada 6,  Duelo de tritones (Capítulo 26)]
279                                                                      [The Break-Up_hook_primary_16x9]
962                                                              [Fuller House,  Temporada 5 (Tráiler 2)]
673                                                                                  [THECALL_CLIP_GENRE]
525                           [Luis Miguel - La serie,  Temporada 1,  Yo que no vivo sin ti (Capítulo 3)]
2717                              [Community,  Temporada 3,  Geografía del conflicto global (Capítulo 2)]
574                                                             [Sonic Boom,  Season 1_hook_primary_16x9]
1444                           [El espectacular Hombre Araña,  Temporada 1,  Group Therapy (Capítulo 11)]
7348  [Ultimate Spider-Man,  Marvel Ultimate Spider-Man vs. Los 6 Siniestros,  Regreso al Univers-arañ...
432                                                                                           [Rocketman]

In [1028]:
#En cuantas partes de dividieron nuestros títulos?
df_sep_lista["parts"] = df_sep_lista["title"].apply(len)

In [1029]:
#Verificamos la creación de la nueva columna
df_sep_lista

title  \
0        [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 15 (Capítulo 15)]   
1        [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 14 (Capítulo 14)]   
2        [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 13 (Capítulo 13)]   
3        [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 12 (Capítulo 12)]   
4                           [Tiger & Bunny,  TIGER & BUNNY,  Season 1_hook_01_16x9]   
5                                   [Better Call Saul,  Season 3_hook_primary_16x9]   
6                                                              [Regreso al espacio]   
7                                                    [Tráiler,  Regreso al espacio]   
...                                                                             ...   
9132                                              [Jurassic World,  Mundo Jurásico]   
9133                                              [Jurassic World,  Mundo Jurásico]   
9134                [Altered Carbon,  Temporada 1,  Retorno al pasado (Capítulo 1)]   
9135       [Sword Art Online,  Sword Art Online II,  Un pequeño paso (Capítulo 14)]   
9136        [Sword Art Online,  Sword Art Online II,  Phantom Bullet (Capítulo 13)]   
9137        [Sword Art Online,  Sword Art Online II,  Phantom Bullet (Capítulo 13)]   
9138         [Sword Art Online,  Sword Art Online II,  Bala fantasma (Capítulo 12)]   
9139  [Sword Art Online,  Sword Art Online II,  El mundo de las armas (Capítulo 1)]   

      parts  
0         3  
1         3  
2         3  
3         3  
4         3  
5         2  
6         1  
7         2  
...     ...  
9132      2  
9133      2  
9134      3  
9135      3  
9136      3  
9137      3  
9138      3  
9139      3  

[9140 rows x 2 columns]

¿Qué nos puede decir esta nueva variable? "parts"

In [1030]:
df_sep_lista["parts"].value_counts()

3    6313
1    1132
4     877
2     785
5      17
6       9
8       4
7       3
Name: parts, dtype: int64

* La mayoría de títulos se han separado en 3 partes, esto es lógico ya que al analizar el formato del título de una serie esta constaba de 3 partes. Sin embargo no siempre es infalible, dado a que cualquiera de los términos puede incluir un ":"
* El segundo en cantidad de incidencias se da en 1 parte, lo cual podrían a venir siendo las pelis, aunque igualmente esto no es infalible, las películas pueden contener ":"
* Lo que explicaría las incidencias de 1 parte hasta un máximo de 8 partes, siendo el de más incidencias el separar en 3 partes, y la menor separarla en 7 partes. 

# Muestreo de Elementos

* Como se ven aquellos elementos que se dividen en 2 partes? en 3 partes? en 1 parte?

In [1031]:
condicion = df_sep_lista["parts"] == 1
print(df_sep_lista[condicion].shape)
df_sep_lista[condicion].sample(17)

(1132, 2)


title  parts
210        [The Vanishing_hook_05_16x9]      1
3004                [Pelea de maestros]      1
804   [Kung Fu Panda_Manualhook_4_16x9]      1
753                      [Chica skater]      1
6948                        [El origen]      1
4967                          [Matilda]      1
6104                          [Aliados]      1
7061                         [Silencio]      1
3969            [La película de Stitch]      1
5574                        [Supercool]      1
1039       [El ejército de los muertos]      1
5960                 [La casa del lago]      1
8105                [Soltera codiciada]      1
2720            [Un refugio inesperado]      1
3429    [Fiebre de sábado por la noche]      1
5377                       [Karate Kid]      1
3944                          [¡Viven!]      1

* Los elementos que no se dividen y siguen íntegros efectivamente son en su mayoría películas, sin embargo también es posible que en menor medida aparezcan elementos catalogados correctamente en la columna suple_type, tales como los Hooks o algún otro material. 
* Son engorrosamente contables, se requiere un análisis con otro enfoque. 

In [1032]:
condicion = df_sep_lista["parts"] == 2
print(df_sep_lista[condicion].shape)
df_sep_lista[condicion].sample(17)

(785, 2)


title  \
661                                  [Temporada 1 (Tráiler),  El camino de la noche]   
1991                                       [Dragon’s Dogma,  Hook 3 (Opening Music)]   
269                                    [The Lost World,  Jurassic Park_hook_01_16x9]   
7285  [American Crime Story,  The Assassination of Gianni Versace_hook_primary_16x9]   
858                                                      [Upgrade,  Máquina asesina]   
8329                                      [Dragon Ball Z,  La Batalla de los Dioses]   
8374                                               [American Pie 2,  Tu segunda vez]   
615                                             [Arrow,  Season 7_hook_primary_16x9]   
7611                           [Una historia de terror,  Halloween con Mickey Mouse]   
1313                                   [Peaky Blinders,  Season 5_hook_primary_16x9]   
7916                                  [Warcraft,  El primer encuentro de dos mundos]   
1311                                   [Grey's Anatomy,  Season 1_hook_primary_16x9]   
1286                                             [Tráiler,  Noticias del gran mundo]   
61                                               [Toradora!,  Season 1_hook_01_16x9]   
7416                                                [Estación zombie,  Tren a Busan]   
1155                                               [Parte 3 (Tráiler),  Paradise PD]   
1365                                         [Sweet Home,  Season 1_CLIP_MUSIC_NON1]   

      parts  
661       2  
1991      2  
269       2  
7285      2  
858       2  
8329      2  
8374      2  
615       2  
7611      2  
1313      2  
7916      2  
1311      2  
1286      2  
61        2  
7416      2  
1155      2  
1365      2

* En un primer vistazo parecen comprobar fácilmente que existen películas con el caracter ":" 
* También podemos visualizar avances pero en específicos aquellos que pertenecen a series, por lo que la variable "suple_types" podría ser importante para ayudarnos. 
* Son engorrosamente contables, se requiere un análisis con otro enfoque. 

In [1033]:
condicion = df_sep_lista["parts"] == 3
print(df_sep_lista[condicion].shape)
df_sep_lista[condicion].sample(17)

(6313, 2)


title  \
2545                  [That '70s Show,  Temporada 2,  El último día de Red (Capítulo 2)]   
652                                   [Seinfeld,  Temporada 2,  La exnovia (Capítulo 1)]   
2771                  [Rick y Morty,  Temporada 3,  Los recuerdos de Morty (Capítulo 8)]   
5727              [Brooklyn Nine-Nine,  Temporada 2,  La defensa descansa (Capítulo 14)]   
1759                           [Gambito de dama,  Miniserie,  Intercambios (Capítulo 2)]   
2648                       [Community,  Temporada 5,  Escapismo geotérmico (Capítulo 5)]   
8647                             [(Des)encanto,  Parte 1,  Vocación mortal (Capítulo 5)]   
5293          [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 14 (Capítulo 14)]   
6376                           [Brooklyn Nine-Nine,  Temporada 3,  La 9-8 (Capítulo 15)]   
598                             [Trabajo incógnito,  Parte 1,  Sangre azul (Capítulo 3)]   
7466                             [MERLÍN,  Temporada 3,  El oro del duende (Capítulo 3)]   
2633         [Community,  Temporada 6,  Decoro básico en la sala de crisis (Capítulo 3)]   
6255                      [Brooklyn Nine-Nine,  Temporada 1,  Pizza de Sal (Capítulo 9)]   
4710                      [Brooklyn Nine-Nine,  Temporada 4,  Halloween IV (Capítulo 5)]   
4550          [La vida desastrosa de Saiki K.,  Temporada 1,  Episodio 11 (Capítulo 11)]   
6881  [Sirius the Jaeger,  Temporada 1,  Un espíritu aúlla en la oscuridad (Capítulo 1)]   
4885                  [Brooklyn Nine-Nine,  Temporada 5,  Noche de juegos (Capítulo 10)]   

      parts  
2545      3  
652       3  
2771      3  
5727      3  
1759      3  
2648      3  
8647      3  
5293      3  
6376      3  
598       3  
7466      3  
2633      3  
6255      3  
4710      3  
4550      3  
6881      3  
4885      3

* En este caso es muy probable que nos encontremos con un título que es en efecto una serie, cabe resaltar también que no todas las series se dividen por temporadas sino que en ocasiones también podría ser por "partes" o alguna otra denominación diegética. 
* Las excepciones son mínimas pero existen. 

In [1034]:
condicion = df_sep_lista["parts"] == 4
print(df_sep_lista[condicion].shape)
df_sep_lista[condicion].sample(17)

(877, 2)


title  \
5838                              [Los 3 de abajo,  Relatos de Arcadia,  Parte 1,  La plaga (Capítulo 4)]   
6878                                                [BAKI,  Temporada 1,  Parte 1,  Dulces (Capítulo 12)]   
5820                    [Los 3 de abajo,  Relatos de Arcadia,  Parte 1,  Contra toda lógica (Capítulo 9)]   
1315  [Escena del crimen,  Desaparición en el hotel Cecil,  Miniserie,  Desaparecida en Los Ángeles (C...   
4342                    [El Dragón,  El regreso de un guerrero,  Temporada 1,  Episodio 33 (Capítulo 33)]   
7257                            [13 de noviembre,  Terror en París,  Miniserie,  Episodio 1 (Capítulo 1)]   
9012                               [Devil May Cry,  Temporada 1,  Misión 6,  Reina del rock (Capítulo 6)]   
3507  [Johnny Test,  Temporada 5,  Johnny invisible / Johnny contra Rorro Pirrorro,  La batalla final ...   
3976  [Mickey,  Aventuras sobre ruedas,  Temporada 1,  ¡Neumático en fuga! / Cuidando a la gatita (Cap...   
5674                         [Trollhunters,  Relatos de Arcadia,  Parte 2,  Juicio anulado (Capítulo 12)]   
5033                            [Trollhunters,  Relatos de Arcadia,  Parte 3,  El juramento (Capítulo 7)]   
5932                    [Trollhunters,  Relatos de Arcadia,  Parte 2,  Entre siseos y tiros (Capítulo 6)]   
7720  [DreamWorks - Cuentos del pantano de Shrek,  Volumen 1,  Un relato de Shrek,  Asústame si puedes...   
998                                                    [Selena,  La serie,  Parte 2,  Ya no (Capítulo 5)]   
1121                                     [Genio del mal,  Temporada 1,  Parte 1,  El atraco (Capítulo 1)]   
4251       [Rick y Morty,  Temporada 4,  Aprovechando el mortymento,  Vive, muere, Rickpeat (Capítulo 1)]   
5253                     [Los 3 de abajo,  Relatos de Arcadia,  Parte 2,  Día de las Madres (Capítulo 4)]   

      parts  
5838      4  
6878      4  
5820      4  
1315      4  
4342      4  
7257      4  
9012      4  
3507      4  
3976      4  
5674      4  
5033      4  
5932      4  
7720      4  
998       4  
1121      4  
4251      4  
5253      4

* Dividido en cuatro partes nuevamente al parecer suele ser común toparse con una serie, sin embargo existe un patrón interesante y es que aquí se encuentras series cuyos títulos contengan ":"
* Engorroso revisar manualmente.

In [1035]:
condicion = df_sep_lista["parts"] == 5
print(df_sep_lista[condicion].shape)
df_sep_lista[condicion].sample(17)

(17, 2)


title  \
4690  [BoJack Horseman,  Temporada 1,  BoJack Horseman,  Temporada 1,  La historia de BoJack, capítulo...   
6357  [Kung Fu Panda,  Secretos bárbaros,  Volumen 1,  Jugar los dos,  Secretos Bárbaros de Kung Fu Pa...   
6277  [Kung Fu Panda,  Secretos bárbaros,  Volumen 1,  Jugar los dos,  Secretos Bárbaros de Kung Fu Pa...   
1630                     [La leyenda de Korra,  Libro dos,  Espíritus,  Comienzos,  Parte 2 (Capítulo 8)]   
6002                    [Trollhunters,  Relatos de Arcadia,  Parte 1,  El elegido,  Parte 2 (Capítulo 2)]   
3459  [Johnny Test,  Temporada 1,  Johnny Test,  Party Monster / Johnny Test,  Extreme Crime Stopper (...   
763   [Johnny Test,  Temporada 1,  Johnny Test,  Party Monster / Johnny Test,  Extreme Crime Stopper (...   
827                     [Trollhunters,  Relatos de Arcadia,  Parte 1,  El elegido,  Parte 1 (Capítulo 1)]   
1631                     [La leyenda de Korra,  Libro dos,  Espíritus,  Comienzos,  Parte 1 (Capítulo 7)]   
6279  [Kung Fu Panda,  Secretos bárbaros,  Volumen 1,  Jugar los dos,  Secretos Bárbaros de Kung Fu Pa...   
6003                    [Trollhunters,  Relatos de Arcadia,  Parte 1,  El elegido,  Parte 1 (Capítulo 1)]   
780   [Johnny Test,  Temporada 1,  Johnny Test,  Party Monster / Johnny Test,  Extreme Crime Stopper (...   
5721  [Johnny Test,  Temporada 1,  Johnny Test,  Party Monster / Johnny Test,  Extreme Crime Stopper (...   
2722        [Community,  Temporada 2,  Community,  Temporada 2,  A Fist Full of Paintballs (Capítulo 22)]   
8822  [Dave Chappelle,  Equanimity & The Bird Revelation,  Colección,  Dave Chappelle,  Equanimity (Ca...   
767   [Johnny Test,  Temporada 1,  Johnny Test,  Party Monster / Johnny Test,  Extreme Crime Stopper (...   
4707  [BoJack Horseman,  Temporada 1,  BoJack Horseman,  Temporada 1,  La historia de BoJack, capítulo...   

      parts  
4690      5  
6357      5  
6277      5  
1630      5  
6002      5  
3459      5  
763       5  
827       5  
1631      5  
6279      5  
6003      5  
780       5  
5721      5  
2722      5  
8822      5  
767       5  
4707      5

* En cinco partes sufrimos la misma condición anterior, con seguridad es una serie, que contiene ":" tanto en el título como en el nombre del episodio. 
* Conclusión son Fácilmente revisables (usar shape), todas en efecto son una serie. Los Divididos en 5 partes corresponden a una serie. 

In [1036]:
condicion = df_sep_lista["parts"] > 5
print(df_sep_lista[condicion].shape)
df_sep_lista[condicion].sample(16)

(16, 2)


title  \
1943  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  El maestro de Sokka (Capí...   
1894  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  El templo Aire del oeste ...   
1891  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  The Boiling Rock,  Parte ...   
1897  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  El avatar y el señor del ...   
4728  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  Sozin's Comet,  Parte 4, ...   
8392  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 1,  El niño en el iceberg (Ca...   
1890  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  The Boiling Rock,  Parte ...   
1896     [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  La fugitiva (Capítulo 7)]   
8313  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 1,  El niño en el iceberg (Ca...   
1898        [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  La playa (Capítulo 5)]   
2144  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 1,  El niño en el iceberg (Ca...   
4729  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  Sozin's Comet,  Parte 3, ...   
4731  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  Sozin's Comet,  Parte 1, ...   
2143        [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 2,  El gurú (Capítulo 19)]   
4730  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  Sozin's Comet,  Parte 2, ...   
1895  [Avatar,  La leyenda de Aang,  Avatar,  la leyenda de Aang,  Libro 3,  The Day of the Black Sun,...   

      parts  
1943      6  
1894      6  
1891      7  
1897      6  
4728      8  
8392      6  
1890      7  
1896      6  
8313      6  
1898      6  
2144      6  
4729      8  
4731      8  
2143      6  
4730      8  
1895      7

* Los títulos con más de 5 partes son escasos y siguen un patron que nos dice que son de un mismo material, Una Serie. 
* Conclusión: Si tenemos una lista está dividida entre 6 y 8 partes es una Serie. 

# 3.- FASE DE IMPLEMENTACIÓN O VARIABLES ADICIONALES

In [1037]:
#Vamos a intentar generalizar, diremos que títulos con dos partes o menos serán películas, y de 3 a mas Series, esto por su estructura. 
df_sep_lista["type"] = np.where(df_sep_lista["parts"]<3,"movie","series") #Hacemos uso de numpy "WHERE" Condición de la serie, y nombres del contenido. 

In [1038]:
#Comprobamos. 
df_sep_lista

title  \
0        [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 15 (Capítulo 15)]   
1        [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 14 (Capítulo 14)]   
2        [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 13 (Capítulo 13)]   
3        [La vida desastrosa de Saiki K.,  Temporada 2,  Episodio 12 (Capítulo 12)]   
4                           [Tiger & Bunny,  TIGER & BUNNY,  Season 1_hook_01_16x9]   
5                                   [Better Call Saul,  Season 3_hook_primary_16x9]   
6                                                              [Regreso al espacio]   
7                                                    [Tráiler,  Regreso al espacio]   
...                                                                             ...   
9132                                              [Jurassic World,  Mundo Jurásico]   
9133                                              [Jurassic World,  Mundo Jurásico]   
9134                [Altered Carbon,  Temporada 1,  Retorno al pasado (Capítulo 1)]   
9135       [Sword Art Online,  Sword Art Online II,  Un pequeño paso (Capítulo 14)]   
9136        [Sword Art Online,  Sword Art Online II,  Phantom Bullet (Capítulo 13)]   
9137        [Sword Art Online,  Sword Art Online II,  Phantom Bullet (Capítulo 13)]   
9138         [Sword Art Online,  Sword Art Online II,  Bala fantasma (Capítulo 12)]   
9139  [Sword Art Online,  Sword Art Online II,  El mundo de las armas (Capítulo 1)]   

      parts    type  
0         3  series  
1         3  series  
2         3  series  
3         3  series  
4         3  series  
5         2   movie  
6         1   movie  
7         2   movie  
...     ...     ...  
9132      2   movie  
9133      2   movie  
9134      3  series  
9135      3  series  
9136      3  series  
9137      3  series  
9138      3  series  
9139      3  series  

[9140 rows x 3 columns]

In [1039]:
df = pd.concat([df,df_sep_lista["type"]],axis= 1)

In [1040]:
df

time_start  duration                                  atributes  \
0      18:38:28  00:13:25                                        NaN   
1      18:13:46  00:20:58  Autoplayed: user action: User_Interaction   
2      13:41:14  00:19:31                                        NaN   
3      13:32:44  00:08:26                                        NaN   
4      13:32:15  00:00:18              Autoplayed: user action: None   
5      13:32:07  00:00:05              Autoplayed: user action: None   
6      23:51:48  00:14:37                                        NaN   
7      23:51:36  00:00:09              Autoplayed: user action: None   
...         ...       ...                                        ...   
9132   22:23:29  00:27:26       Autoplayed: user action: Unspecified   
9133   22:18:49  00:04:33       Autoplayed: user action: Unspecified   
9134   21:31:51  00:03:02       Autoplayed: user action: Unspecified   
9135   21:21:51  00:02:49                                        NaN   
9136   21:04:49  00:14:54       Autoplayed: user action: Unspecified   
9137   21:03:12  00:01:32                                        NaN   
9138   20:42:08  00:20:54                                        NaN   
9139   20:41:19  00:00:02              Autoplayed: user action: None   

                                                                          title  \
0        La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
1        La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
2        La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
3        La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
4                           Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   
5                                  Better Call Saul: Season 3_hook_primary_16x9   
6                                                            Regreso al espacio   
7                                                   Tráiler: Regreso al espacio   
...                                                                         ...   
9132                                             Jurassic World: Mundo Jurásico   
9133                                             Jurassic World: Mundo Jurásico   
9134                Altered Carbon: Temporada 1: Retorno al pasado (Capítulo 1)   
9135       Sword Art Online: Sword Art Online II: Un pequeño paso (Capítulo 14)   
9136        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
9137        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
9138         Sword Art Online: Sword Art Online II: Bala fantasma (Capítulo 12)   
9139  Sword Art Online: Sword Art Online II: El mundo de las armas (Capítulo 1)   

     suple_type                                    device       date    type  
0           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
1           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
2           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
3           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
4          HOOK  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
5          HOOK  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11   movie  
6           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-10   movie  
7       TRAILER  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-10   movie  
...         ...                                       ...        ...     ...  
9132        NaN                                  Sony PS4 2018-05-26   movie  
9133        NaN                                  Sony PS4 2018-05-26   movie  
9134        NaN                                  Sony PS4 2018-05-23  series  
9135        NaN                                  Sony PS4 2018-05-23  series  
9136        NaN                                  Sony PS4 2018-05-23  series  
9137        NaN             

* La importancia de Usar .loc

In [1041]:
#Cambiamos los valores de nuestra columna "type" según la condición de otra "suple_type"
df.loc[df["suple_type"].notna(),'type'] = 'bonus' #Si, suple_type no es nulo o NAN, 

In [1042]:
#Comprobamos y en efecto, tenemos lo deseado. 
df #De tener un atributo diferente a NAN en la variable suplemento, el tipo será bonus. 

time_start  duration                                  atributes  \
0      18:38:28  00:13:25                                        NaN   
1      18:13:46  00:20:58  Autoplayed: user action: User_Interaction   
2      13:41:14  00:19:31                                        NaN   
3      13:32:44  00:08:26                                        NaN   
4      13:32:15  00:00:18              Autoplayed: user action: None   
5      13:32:07  00:00:05              Autoplayed: user action: None   
6      23:51:48  00:14:37                                        NaN   
7      23:51:36  00:00:09              Autoplayed: user action: None   
...         ...       ...                                        ...   
9132   22:23:29  00:27:26       Autoplayed: user action: Unspecified   
9133   22:18:49  00:04:33       Autoplayed: user action: Unspecified   
9134   21:31:51  00:03:02       Autoplayed: user action: Unspecified   
9135   21:21:51  00:02:49                                        NaN   
9136   21:04:49  00:14:54       Autoplayed: user action: Unspecified   
9137   21:03:12  00:01:32                                        NaN   
9138   20:42:08  00:20:54                                        NaN   
9139   20:41:19  00:00:02              Autoplayed: user action: None   

                                                                          title  \
0        La vida desastrosa de Saiki K.: Temporada 2: Episodio 15 (Capítulo 15)   
1        La vida desastrosa de Saiki K.: Temporada 2: Episodio 14 (Capítulo 14)   
2        La vida desastrosa de Saiki K.: Temporada 2: Episodio 13 (Capítulo 13)   
3        La vida desastrosa de Saiki K.: Temporada 2: Episodio 12 (Capítulo 12)   
4                           Tiger & Bunny: TIGER & BUNNY: Season 1_hook_01_16x9   
5                                  Better Call Saul: Season 3_hook_primary_16x9   
6                                                            Regreso al espacio   
7                                                   Tráiler: Regreso al espacio   
...                                                                         ...   
9132                                             Jurassic World: Mundo Jurásico   
9133                                             Jurassic World: Mundo Jurásico   
9134                Altered Carbon: Temporada 1: Retorno al pasado (Capítulo 1)   
9135       Sword Art Online: Sword Art Online II: Un pequeño paso (Capítulo 14)   
9136        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
9137        Sword Art Online: Sword Art Online II: Phantom Bullet (Capítulo 13)   
9138         Sword Art Online: Sword Art Online II: Bala fantasma (Capítulo 12)   
9139  Sword Art Online: Sword Art Online II: El mundo de las armas (Capítulo 1)   

     suple_type                                    device       date    type  
0           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
1           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
2           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
3           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11  series  
4          HOOK  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11   bonus  
5          HOOK  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-11   bonus  
6           NaN  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-10   movie  
7       TRAILER  Sony Sony Android TV 2018 M2 4K Smart TV 2022-04-10   bonus  
...         ...                                       ...        ...     ...  
9132        NaN                                  Sony PS4 2018-05-26   movie  
9133        NaN                                  Sony PS4 2018-05-26   movie  
9134        NaN                                  Sony PS4 2018-05-23  series  
9135        NaN                                  Sony PS4 2018-05-23  series  
9136        NaN                                  Sony PS4 2018-05-23  series  
9137        NaN             

# Realizamos una pequeña comprobación con lo que respecta a las series. 

In [1067]:
condicion = df["type"] == "series"
df[condicion].sample(17)

time_start  duration                                  atributes  \
229    15:01:15  00:17:31                                        NaN   
3185   14:22:28  00:14:30              Autoplayed: user action: None   
5904   21:54:47  00:22:07              Autoplayed: user action: None   
3369   20:05:50  00:24:18              Autoplayed: user action: None   
1224   13:48:12  00:00:34       Autoplayed: user action: Unspecified   
6139   23:50:42  00:00:45       Autoplayed: user action: Unspecified   
5034   19:03:27  00:03:15                                        NaN   
334    12:58:04  00:00:09  Autoplayed: user action: User_Interaction   
1796   18:38:38  00:00:06              Autoplayed: user action: None   
5807   17:30:37  00:00:08       Autoplayed: user action: Unspecified   
3587   16:28:16  00:21:11       Autoplayed: user action: Unspecified   
8097   02:12:10  00:01:31                                        NaN   
6135   00:27:29  00:10:42                                        NaN   
6346   03:34:06  00:22:14                                        NaN   
4756   00:29:00  00:21:18       Autoplayed: user action: Unspecified   
8843   22:29:54  00:49:52                                        NaN   
8984   22:02:46  00:18:57              Autoplayed: user action: None   

                                                                                                    title  \
229        Yu-Gi-Oh!: Temporada 2: Doble duelo: Yugi y Kaiba contra Lumis y Umbra (Parte 4) (Capítulo 73)   
3185                                         El final del paraíso: Temporada 1: De rodillas (Capítulo 66)   
5904             Trollhunters: Relatos de Arcadia: Parte 3: Así que, salgo con una hechicera (Capítulo 4)   
3369                                                     Pandemia: Temporada 1: Nos persigue (Capítulo 1)   
1224                                         Community: Temporada 2: Mixology Certification (Capítulo 10)   
6139                                         Sonic Boom: Temporada 1: Venganza de Año Nuevo (Capítulo 42)   
5034                                 Trollhunters: Relatos de Arcadia: Parte 3: El juramento (Capítulo 7)   
334                                                 Cobra Kai: Temporada 3: 19 de diciembre (Capítulo 10)   
1796                        Flash: Temporada 6: La última tentación de Barry Allen (Parte 1) (Capítulo 7)   
5807                              Trollhunters: Relatos de Arcadia: Parte 2: Solo añade agua (Capítulo 8)   
3587                                           Brooklyn Nine-Nine: Temporada 4: Halloween IV (Capítulo 5)   
8097  The Seven Deadly Sins: Resurrección de los Diez Mandamientos: El regreso de los pecadores (Capít...   
6135                                              Sonic Boom: Temporada 1: El amor de Tails (Capítulo 46)   
6346                                      Brooklyn Nine-Nine: Temporada 3: Máxima seguridad (Capítulo 21)   
4756                                                   Flash: Temporada 5: La muerte de Vibe (Capítulo 3)   
8843                                              Good Morning Call: Temporada 1: Nochebuena (Capítulo 4)   
8984                                    Yu-Gi-Oh!: Temporada 5: Un paso adelante (Parte 2) (Capítulo 195)   

     suple_type                                          device       date  \
229         NaN        Sony Sony Android TV 2018 M2 4K Smart TV 2022-02-14   
3185        NaN        Sony Sony Android TV 2018 M2 4K Smart TV 2020-04-14   
5904        NaN                                        Sony PS4 2019-06-15   
3369        NaN        Sony Sony Android TV 2018 M2 4K Smart TV 2020-03-25   
1224        NaN        Sony Sony Android TV 2018 M2 4K Smart TV 2021-02-26   
6139        NaN                                        Sony PS4 2019-05-16   
5034        NaN    Android DefaultWidevineL3Phone Android Phone 2019-09-22   
334         NaN        Sony Sony Android TV 2018 M2 4K Smart TV 2021-12-31   
1796        NaN        Sony Sony Android TV

* En efecto parece que la hemos catalogado de forma satisfactoria, quizás exista algúna excepción, sin embargo tras un muestreo es evidente que este método es efectivo. 
* Sobre todo tener en cuenta la variable suple_type fue de vital importancia
* Regla: Lista de dos a menos partes es una película, de lo contrario, la catalogamos como serie. RECORDAR QUE USAMOE NUMPY.WHERE